### ISSUE TITLE
Panic at startup

### ISSUE THREAD

**Issue by killkrt** [Wumpf/blub#47]
**Software version**:
3f5be1e657513ec4fb2a36b47d9487ff5a7d6507

**Issue:**
Panic at startup (executing in release mode) with the following error:
```thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Decoding(DecodingError { format: Exact(Hdr), underlying: Some(RadianceHdrSignatureInvalid) })', src/renderer/background.rs:110:68```

In debug mode I have this different error:
```
ERROR wgpu::backend::direct] wgpu error: Validation Error

    Caused by:
        In Device::create_texture
          note: label = 'Cubemap'
        Dimension X is zero

thread main panicked at Handling wgpu errors as fatal by default
```

**System details:**
OS: Arch Linux 64 bit (Kernel 5.18.16)
GCC: 12.1.1
Windows manager: i3
Hardware: NUC8I5BEH2
Rust: 1.62.1

**Comment by anadb** [comment]
same issue, did you figure it out?

**Comment by killkrt** [comment]
No, I gave up, the project seems to be abandoned, though.

**Comment by Wumpf** [comment]
yeah I haven't touched anything here in ages. The wgpu version used is beyond ancient and there's too many things I'd do differently nowadays

**Comment by jmatsushita** [comment]
> yeah I haven't touched anything here in ages. The wgpu version used is beyond ancient and there's too many things I'd do differently nowadays

@Wumpf just watched your awesome presentation on the rust game dev youtube channel https://youtu.be/Yzr9va5UtiE?si=q_XSB5LV_G1fawa7 

Really curious about how you would approach this nowadays! 

Cheers :)

**Comment by Wumpf** [comment]
@jmatsushita thank you :)

Hmm where to start.... it's been so long ago, but here's a few things off the top of my head:
* stop using volume textures: there's some merit to them for _potentially_ fast sampling & filtering, but it's a portability nightmore. Not just the capabilities aren't properly portable, also the performance characteristics are unpredictable
* use something else entirely for boundary conditions: everything I tried was either super hard to implement & understand or super leaky. Might even consider throwing cpu processing into the mix.
* while I fiddled around with blub, MLS-MPM become quite popular and there's a lot of implementations available to learn from, definitely work a deeper look
* less algorithmically on the "would do differently" side I just learned a lot more about Rust over the years so I'm likely to structure things differently. Also, I find a proper cross-platform working version more interesting, so I'd strongly consider sticking to the WebGPU feature set, allowing to run everything in the browser
